### ISSUE TITLE
[3.0 regression] Cursor specific test run broken for classes, namespaces and files

### ISSUE THREAD

**Issue by cremor** [microsoft/testfx#1564]
## Describe the bug
After updating MSTest from 2.2.10 to 3.0.2 I can't execute all tests in a test class, namespace or file any more. It always executes 0 tests.

## Steps To Reproduce
1. Create a new test project. (It will be created with MSTest.TestFramework 2.2.10.)
2. Add 3 test classes with 2 test methods each. One test class should be in its own file, another file should contain 2 test classes.
3. Place the cursor at the following locations and execute "Run Tests" from the context menu. Check how many tests were executed in the status bar.
    - In a test method: 1 test is executed
    - In a test class (outside of methods): 2 tests are executed
    - In a namespace (outside of classes): 6 tests are executed
    - In a file (ouside of namespaces): 2 or 4 tests are executed, depending on the file
4. Update MSTest.TestFramework to 3.0.2
5. Run the checks from step 3 again.

## Expected behavior
It should execute the same amount of tests as with the old version.

## Actual behavior
Only execution inside a test method works. For a test classes, namespaces and files it always executes 0 tests.

[AB#1754963](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1754963)

**Comment by Evangelink** [comment]
Hey @cremor,

I just tried to reproduce the issue and file, namespace, class and method context run are working fine for me (see below gif):

![Run method-class-namespace](https://user-images.githubusercontent.com/11340282/214013736-79641e5e-db82-4b1e-8a42-b1e9dff4a13c.gif)

Here is my setup:

`Test12.csproj`:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>

    <IsPackable>false</IsPackable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
    <PackageReference Include="MSTest.TestAdapter" Version="3.0.2" />
    <PackageReference Include="MSTest.TestFramework" Version="3.0.2" />
    <PackageReference Include="coverlet.collector" Version="3.1.2" />
  </ItemGroup>

</Project>
```

`UnitTest1.cs`:

```csharp
namespace TestProject12 {
    [TestClass]
    public class UnitTest1 {
        [TestMethod]
        public void TestMethod1() {
            
        }
        [TestMethod]
        public void TestMethod2() {
        }
    }
}
```

`UnitTest2And3.cs`:

```csharp
namespace TestProject12 {
    [TestClass]
    public class UnitTest2 
    {
        [TestMethod]
        public void TestMethod1() {
        }
        [TestMethod]
        public void TestMethod2() {
        }
    }

    [TestClass]
    public class UnitTest3 
    {
        [TestMethod]
        public void TestMethod1() {

        }
        [TestMethod]
        public void TestMethod2() {
        }
    }
}

```

**Comment by cremor** [comment]
This is weird. My test solution is nearly the same. Even if I change it to be exactly like your code I can still reproduce the issue.

Can you try it with my solution? [TestProject1.zip](https://github.com/microsoft/testfx/files/10478988/TestProject1.zip)
Here is the tests output for me:

```
Building Test Projects
Executing test method: TestProject1.UnitTest1.TestMethod1
========== Starting test run ==========
========== Test run finished: 1 Tests (1 Passed, 0 Failed, 0 Skipped) run in 40 ms ==========
Building Test Projects
Executing all tests in type: TestProject1.UnitTest1
========== Starting test run ==========
========== Test run finished: 0 Tests (0 Passed, 0 Failed, 0 Skipped) run in < 1 ms ==========
Building Test Projects
Executing all tests in namespace: TestProject1
========== Starting test run ==========
========== Test run finished: 0 Tests (0 Passed, 0 Failed, 0 Skipped) run in < 1 ms ==========
Building Test Projects
Executing all tests in file: C:\Daten\Projekte\TestProject1\TestProject1\UnitTest1.cs
========== Starting test run ==========
========== Test run finished: 0 Tests (0 Passed, 0 Failed, 0 Skipped) run in < 1 ms ==========
```

The `Executing all tests in x` part is always correct. But it just doesn't find any tests.

I'm using Visual Studio version 17.4.4

**Comment by Evangelink** [comment]
Oh I am sorry, I did my test with my dogfood version! You are right that there is some issue when running with 17.4.4! 

If you look at the output you will see that there are some warnings that relates to this issue: https://github.com/microsoft/vstest/issues/4208 (marked as closed because it's fixed in 17.5 preview). Could you confirm you also see these warnings?

**Comment by cremor** [comment]
Which warning do you mean and where should I see it?
I don't see any warnings in the test explorer or in the build or tests output panes.

**Comment by Evangelink** [comment]
Interesting in the `Output` `Tests` I do see the error (screenshot below):
![image](https://user-images.githubusercontent.com/11340282/214036268-d6db8c09-9a7f-47bb-aeb8-edc2d113283c.png)

**Comment by cremor** [comment]
I do not get that exception in the output. My full output is the one I copied to the previous comment.

**Comment by peterwald** [comment]
@Evangelink, the bug fix referenced in https://github.com/microsoft/vstest/issues/4208 has already been ported to `17.4` and will ship in `17.4.5`.

**Comment by cremor** [comment]
@Evangelink @peterwald I can still reproduce this bug with VS 17.4.5 (which was just released).

**Comment by Evangelink** [comment]
Reopening for investigation (cc @engyebrahim)

**Comment by engyebrahim** [comment]
@peterwald @Evangelink 
I still can reproduce the issue in `v17.4.5`.
but in the preview version `17.5` it's solved.

**Comment by Evangelink** [comment]
@AbhitejJohn @peterwald Looks like the issue on Test Explorer side is still not fixed.

**Comment by cremor** [comment]
Seems like this is fixed in 17.5

**Comment by Evangelink** [comment]
Thanks for the update @cremor
